MeesGame/Gameplay/GameObjects/Tile.cs

```diff
@@ -30,7 +35,7 @@ protected Tile(TileType tileType = TileType.Floor, int layer = 0, string id = ""
                 secondarySprite = new SpriteSheet(GetAssetNamesFromTileType(tileType)[1]);
         }
 
-        public bool Revealed
+        public int Revealed
```

Comment:
Change this to apply the bitwise-OR within the setter. The get function isn't needed, so you can remove that.